pkg/controllers/dashboardapi/settings/settings_test.go

```diff
@@ -257,9 +252,56 @@ func populateTestCases() []*testCase {
 		}
 		test.existingSetting.Name = settingName
 	}
+
 	return testCases
 }
 
-func stringPtr(val string) *string {
-	return &val
+func TestMarkSettingAsUnknownRetry(t *testing.T) {
+	store := map[string]v3.Setting{
+		"unknown": {
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "unknown",
+			},
+			Value:   "unknown",
+			Default: "unknown",
+		},
+	}
+
+	groupResource := schema.GroupResource{
+		Group:    management.GroupName,
+		Resource: v3.SettingResourceName,
+	}
+	client := fake.NewMockNonNamespacedControllerInterface[*v3.Setting, *v3.SettingList](gomock.NewController(t))
+
+	client.EXPECT().Get(gomock.Any(), gomock.Any()).DoAndReturn(func(name string, options metav1.GetOptions) (*v3.Setting, error) {
+		val, ok := store[name]
+		if !ok {
+			return nil, apierrors.NewNotFound(groupResource, name)
+		}
+
+		return &val, nil
+	}).Times(1)
+
+	var updateRun int
+	client.EXPECT().Update(gomock.Any()).DoAndReturn(func(s *v3.Setting) (*v3.Setting, error) {
+		defer func() { updateRun++ }()
+
+		if updateRun == 0 { // Fail the the first update to force retry.
+			return nil, apierrors.NewConflict(groupResource, s.Name, fmt.Errorf("some error"))
+		}
+
+		store[s.Name] = *s
+		return s, nil
+	}).Times(2)
+
+	provider := settingsProvider{
+		settings: client,
+	}
+
+	unknown := store["unknown"]
+
+	err := provider.markSettingAsUnknown(&unknown)
+	assert.Nil(t, err)
+	assert.NotNil(t, store["unknown"].Labels)
+	assert.Equal(t, store["unknown"].Labels["cattle.io/unknown"], "true")
```

Comment:
Nit: Since your test cases already use it, you may want to consider using the const here:

```suggestion
	assert.Equal(t, store["unknown"].Labels[unknownSettingLabelKey], "true")
```